runelite-client/src/main/java/net/runelite/client/plugins/deathindicator/DeathIndicatorPlugin.java

```diff
@@ -0,0 +1,267 @@
+	{
+		try
+		{
+			synchronized (ImageIO.class)
+			{
+				WORLDMAP_ARROW = ImageIO.read(DeathIndicatorPlugin.class.getResourceAsStream("worldmap_arrow.png"));
+				BONES = ImageIO.read(DeathIndicatorPlugin.class.getResourceAsStream("bones.png"));
+			}
+		}
+		catch (IOException e)
+		{
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Provides
+	DeathIndicatorConfig deathIndicatorConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(DeathIndicatorConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		client.clearHintArrow();
+
+		if (worldMapPointSet)
+		{
+			worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+			worldMapPointSet = false;
+		}
+
+		if (deathTimer != null)
+		{
+			infoBoxManager.removeInfoBox(deathTimer);
+		}
+	}
+
+	@Subscribe
+	public void onActorDeath(ActorDeath event)
+	{
+		if (client.isInInstancedRegion())
+		{
+			return;
+		}
+
+		Actor actor = event.getActor();
+
+		if (actor instanceof NPC || actor != client.getLocalPlayer())
+		{
+			return;
+		}
+
+		hasDied = true;
+		playerHasRespawned = false;
+
+		deathLocation = actor.getWorldLocation();
+		deathWorld = client.getWorld();
+
+		if (config.showDeathHintArrow())
+		{
+			client.setHintArrow(deathLocation);
+		}
+
+		if (config.showDeathOnWorldMap())
+		{
+			addMapPoint(deathLocation);
+		}
+
+		if (deathTimer != null)
+		{
+			infoBoxManager.removeInfoBox(deathTimer);
+		}
+
+		if (config.showDeathInfoBox())
+		{
+			deathTimer = new Timer(60, ChronoUnit.MINUTES, BONES, plugin);
+			deathTimer.setTooltip("Died on world: " + deathWorld);
+			infoBoxManager.addInfoBox(deathTimer);
+		}
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (event.getMessage().equals("Oh dear, you are dead!"))
+		{
+			playerHasRespawned = true;
+		}
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		if (!hasDied)
+		{
+			return;
+		}
+
+		if (!playerHasRespawned)
+		{
+			return;
+		}
+
+		currentWorld = client.getWorld();
+
+		if (currentWorld != deathWorld)
+		{
+			return;
+		}
+
+		Player player = client.getLocalPlayer();
+
+		currentLocation = player.getWorldLocation();
+
+		if ((currentLocation.getX() == deathLocation.getX())
+				&& (currentLocation.getY() == deathLocation.getY())
+				&& (currentLocation.getPlane() == deathLocation.getPlane())
+				&& (currentLocation.getRegionID() == deathLocation.getRegionID()))
+		{
+			client.clearHintArrow();
+			infoBoxManager.removeInfoBox(deathTimer);
+			worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+			worldMapPointSet = false;
+			deathMarkerReset = true;
+		}
+
+		if (deathMarkerReset)
+		{
+			hasDied = false;
+			playerHasRespawned = true;
+			deathMarkerReset = false;
+		}
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (event.getGroup().equals("deathindicator") && !config.showDeathHintArrow())
```

Comment:
Maybe I am being picky, but it seems like you should only check for deathindicator changes once. Then inside that if check each config.

Reply:
So like a nested if, where the outter if (event.getGroup().equals("deathindicator")) then the inner if's being if (!config.BLAHBLAHBLAH)? I could get behind that.

Reply:
Yes

Reply:
Will change soon, working on getting the death information saved via hidden config items and upon login, if the player has died and not returned to their items, showing the death information. The current version I've PR'd here does this upon logout/hopping, but on client crash the player is sol atm. If I can't get the saving information and pulling it back up on relaunch to work properly, I'll just update the previous version to meet your request and push that. :)

Reply:
If you need an example of hidden config I think the farming plugin utilizes this.

Reply:
Fixed, thanks for the tip :)